### ISSUE TITLE
Align with core implementation of page header

### ISSUE THREAD

**Issue by matthewcarleton** [patternfly/patternfly-react#2398]
Currently the react version of `.pf-c-header` uses `pf-c-page__header-tools-group` to layout out the utilities. 
In Core it uses `pf-l-toolbar` to layout the items.
This difference is causing some alignment issues.
This could be related to this issue #2009

**Comment by boaz0** [comment]
Hi @matthewcarleton,
I can't see `pf-c-page__header-tools-group` in any of the examples (maybe you meant `pf-c-page__header-tools`?) can you give more information?

**Comment by matthewcarleton** [comment]
Hey @boaz0 ya for sure.
Actually I got it backwards, let me try again :) So in the React implementation we have `pf-l-toolbar` laying out the toolbar in the header. 
In core we don't have the toolbar layout provided at all and use `pf-c-page__header-tools`.
I believe we could update React and Core to use `pf-c-toolbar` but I might be wrong. @mattnolting WDYT?

**Comment by mattnolting** [comment]
@matthewcarleton @boaz0 We definitely need to remove `pf-l-toolbar` from the react implementation, as it's no longer supported in Core. My thought is that we use one standardized layout for flex things like this, but not all of the `pf-l-flex` layout is needed and we can't adjust spacing w/`pf-l-split`. There's an open issue https://github.com/patternfly/patternfly-next/issues/1979 for splitting `pf-l-flex` into a base layout, including responsive spacers, with modifiers being extension/s. That may be the path forward, but until we have a better idea of how that will look, Core's implementation is what we should follow.
